Author: David Palacio <dpalacio@orbitalibre.org>
Description: Make the name of the shared library configurable to use more than one Ruby version
Last-Update: 2011-09-23
Forwarded: no

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,10 +27,10 @@
 
 # Use 'MODULE' instead of 'SHARED' so that qtruby will be built as a bundle on Mac OS X
 # instead of a dyld. But is this correct (rjd)? It needs testing.
-add_library(qtruby4shared SHARED ${qtrubyshared_LIB_SRCS})
+add_library(${QTRUBY4SHAREDNAME} SHARED ${qtrubyshared_LIB_SRCS})
 add_library(qtruby4 MODULE ${qtruby_LIB_SRCS})
 
-target_link_libraries(  qtruby4shared 
+target_link_libraries(  ${QTRUBY4SHAREDNAME}
                             ${QT_QTCORE_LIBRARY} 
                             ${QT_QTGUI_LIBRARY} 
                             ${RUBY_LIBRARY} 
@@ -59,14 +59,14 @@
                             ${SMOKE_QTNETWORK_LIBRARY} 
                             ${SMOKE_QTDBUS_LIBRARY} 
                             ${SMOKE_QTSVG_LIBRARY}
-                            qtruby4shared )
+                            ${QTRUBY4SHAREDNAME} )
 
 # target_link_libraries(qtrubyinternal ${QT_QTCORE_LIBRARY} ${RUBY_LIBRARY} ${QT_QTNETWORK_LIBRARY})
 
 set_target_properties(qtruby4 PROPERTIES PREFIX "")
-set_target_properties(qtruby4shared PROPERTIES VERSION 2.0.0 SOVERSION 2)
+set_target_properties(${QTRUBY4SHAREDNAME} PROPERTIES VERSION 2.0.0 SOVERSION 2)
 
-install(TARGETS qtruby4shared EXPORT QtRuby4Export DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
+install(TARGETS ${QTRUBY4SHAREDNAME} EXPORT QtRuby4Export DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
 install(TARGETS qtruby4 DESTINATION ${CUSTOM_RUBY_SITE_ARCH_DIR})
 install(EXPORT QtRuby4Export DESTINATION ${CMAKE_INSTALL_PREFIX}/share/qtruby4/cmake NAMESPACE QTRUBY4_ )
 
